cloud/disk_manager/internal/pkg/services/pools/storage/storage_ydb_impl.go

```diff
@@ -1348,15 +1353,70 @@ func (s *storageYDB) relocateOverlayDiskTx(
 		return RebaseInfo{}, err
 	}
 
-	if !isPoolConfigured {
-		return RebaseInfo{}, nil
-	}
-
 	freeBaseDisks, err := s.getFreeBaseDisks(ctx, tx, imageID, targetZoneID)
 	if err != nil {
 		return RebaseInfo{}, err
 	}
 
+	// If pool is configured baseDisks will be eventually created.
+	// Else we should create it manually.
+	if len(freeBaseDisks) == 0 && !isPoolConfigured {
+		acquiredBaseDisk, err := s.findBaseDisk(ctx, tx, acquiredSlot.baseDiskID)
+		if err != nil {
+			return RebaseInfo{}, err
+		}
+
+		baseDisk := s.generateBaseDisk(
+			imageID,
+			targetZoneID,
+			acquiredBaseDisk.imageSize,
+			&types.Disk{
+				ZoneId: acquiredBaseDisk.zoneID,
+				DiskId: acquiredBaseDisk.id,
+			},
+		)
+		logging.Info(
+			ctx,
+			"generated base disk: %+v",
+			baseDisk,
+		)
+
+		err = acquireTargetUnitsAndSlots(ctx, tx, &baseDisk, acquiredSlot)
+		if err != nil {
+			return RebaseInfo{}, err
+		}
+
+		acquiredSlotOldState := *acquiredSlot
+
+		acquiredSlot.generation += 1
+		acquiredSlot.targetZoneID = baseDisk.zoneID
+		acquiredSlot.targetBaseDiskID = baseDisk.id
+
+		err = s.updateBaseDiskAndSlot(
+			ctx,
+			tx,
+			baseDiskTransition{
+				oldState: nil,
+				state:    &baseDisk,
+			},
+			slotTransition{
+				oldState: &acquiredSlotOldState,
+				state:    acquiredSlot,
+			},
+		)
+		if err != nil {
+			return RebaseInfo{}, err
+		}
+
+		err = tx.Commit(ctx)
+		if err != nil {
+			return RebaseInfo{}, err
+		}
+
+		// Should wait until base disk is ready.
+		return RebaseInfo{}, errors.NewInterruptExecutionError()
```

Comment:
Why don't we call overlayDiskRebasingTx?

as here https://github.com/ydb-platform/nbs/blob/f7855dff20e228af7443c80fe3d6e8d25034463a/cloud/disk_manager/internal/pkg/services/pools/storage/storage_ydb_impl.go#L1437